torch/utils/show_pickle.py

```diff
@@ -102,7 +103,7 @@ def dump(cls, in_stream, out_stream):
         return value
 
 
-def main(argv, output_stream=None):
+def _main(argv, output_stream=None):
```

Comment:
@dreiss was `main` [intended](https://github.com/pytorch/pytorch/commit/a054d05707f3a5a54e799b82fd95c902a7a8d746) to be a part of the public API? I am asking because I see a failing test with this change:
```
 ======================================================================
2022-09-21T11:22:53.5369137Z ERROR [0.004s]: test_scripted_model (__main__.TestShowPickle)
2022-09-21T11:22:53.5369811Z ----------------------------------------------------------------------
2022-09-21T11:22:53.5370364Z Traceback (most recent call last):
2022-09-21T11:22:53.5370760Z   File "/var/lib/jenkins/workspace/test/test_show_pickle.py", line 29, in test_scripted_model
2022-09-21T11:22:53.5371224Z     torch.utils.show_pickle.main(["", tmp.name + "@*/data.pkl"], output_stream=buf)
2022-09-21T11:22:53.5371769Z AttributeError: module 'torch.utils.show_pickle' has no attribute 'main'
2022-09-21T11:22:53.5372030Z 
2022-09-21T11:22:53.5372766Z ----------------------------------------------------------------------
2022-09-21T11:22:53.5373301Z Ran 4 tests in 1.465s
```

Reply:
Adding it to `__all__` as per @dreiss 's suggestion (on the chat). 

@dreiss please make sure to documentation for it as a follow up!